packages/editor-ui/src/views/NodeView.vue

```diff
@@ -3228,12 +3227,13 @@ export default defineComponent({
 
 				await this.addNodes([newNodeData], [], true);
 
-				const pinData = this.workflowsStore.pinDataByNodeName(nodeName);
-				if (pinData) {
-					this.workflowsStore.pinData({
-						node: newNodeData,
-						data: pinData,
-					});
+				const pinDataForNode = this.workflowsStore.pinDataByNodeName(
+					nodeName,
+				) as INodeExecutionData[];
+				if (pinDataForNode?.length) {
+					try {
+						this.setPinData(newNodeData, pinDataForNode, 'duplicate-node');
+					} catch (error) {}
```

Comment:
```suggestion
					} catch {}
```

Reply:
I'd expect pin data to always be valid on node duplication? It should not error so we should not need to catch?

Reply:
It could error if the maximum size is exceeded when duplicating.